datafusion/core/src/scalar.rs

```diff
@@ -377,7 +377,7 @@ mod tests {
         #[cfg(target_arch = "aarch64")]
         assert_eq!(std::mem::size_of::<ScalarValue>(), 64);
 
-        #[cfg(target_arch = "amd64")]
+        #[cfg(not(target_arch = "aarch64"))]
```

Comment:
I like this formulation as it ensures that one of these `assert_eq!` calls will always be invoked, regardless of architecture

Reply:
You could also write something like

```
let expected = match cfg!(target_arch == "aarch64") {
  true => 64,
  false => 48,
};
```

Reply:
```
        let expected = match cfg!(target_arch == "aarch64") {
            true => 64,
            false => 48,
        };
```

resulted in 

```
error: expected 1 cfg-pattern
   --> datafusion/core/src/scalar.rs:380:30
    |
380 |         let expected = match cfg!(target_arch == "aarch64") {
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `cfg` (in Nightly builds, run with -Z macro-backtrace for more info)

``` 

for me locally

Reply:
Sorry, try removing the double equals...